cmake/tpl/FindMAGMA.cmake

```diff
@@ -70,19 +70,35 @@ if(MAGMA_LIBRARY AND MAGMA_INCLUDE_DIR)
       if(NOT (lib STREQUAL "-lmagma" OR lib STREQUAL "-lmagma_sparse"
             OR lib STREQUAL "-L\${libdir}" OR lib STREQUAL "") )
 
-        # Remove -l only from the beginning of the string
-        string(REPLACE "^-l" "" lib ${lib})
-        list(APPEND _interface_libraires ${lib})
-
         # Check if we need to find roc::hipblas or roc::hipsparse
         if(SUNDIALS_MAGMA_BACKENDS MATCHES "HIP")
-          if((lib STREQUAL "roc::hipblas") AND (NOT TARGET roc::hipblas))
+          if(NOT TARGET roc::hipblas)
             find_package(hipblas REQUIRED)
           endif()
-          if((lib STREQUAL "roc::hipsparse") AND (NOT TARGET roc::hipsparse))
+          if(NOT TARGET roc::hipsparse)
             find_package(hipsparse REQUIRED)
           endif()
+          # MAGMA does not reliably include these in the pkgconfig file
+          set(lib "roc::hipblas;roc::hipsparse")
+        endif()
```

Comment:
Taking a last look over this PR and had a thought. Should this block to add `roc::` targets be outside the for loop since it overwrites the value of `lib` without checking what `lib` is like the CUDA case? Wondering a dependency might unintentionally get lost with this inside the for loop.